agent/http.go

```diff
@@ -194,6 +201,12 @@ func (s *HTTPServer) wrap(handler func(resp http.ResponseWriter, req *http.Reque
 			fmt.Fprint(resp, errMsg)
 		}
 
+		if s.blacklist.IsDisallowed(req.URL.Path) {
+			err := fmt.Errorf("Permission denied, endpoint is disabled")
+			handleErr(err)
```

Comment:
This relying on a string match to make it return an appropriate status code like 403 is weird and seems fragile. Would be nice to make some predefined Err500, Err404, Err403 etc.